apps/emqx_resource/include/emqx_resource_utils.hrl

```diff
@@ -14,10 +14,24 @@
 %% limitations under the License.
 %%--------------------------------------------------------------------
 
+-include_lib("emqx/include/logger.hrl").
+
 -define(SAFE_CALL(_EXP_),
     ?SAFE_CALL(_EXP_, {error, {_EXCLASS_, _EXCPTION_, _ST_}})
 ).
 
+-define(SAFE_CALL_WITH_LOG(_EXP_),
+    ?SAFE_CALL(
+        _EXP_,
+        ?SLOG(warning, #{
+            msg => "safe call error",
```

Comment:
Nit: maybe provide a message as an argument?

Reply:
and we use snake_case_message for `msg`

Reply:
Fixed